app/uk/gov/hmrc/exports/models/declaration/ExportsDeclaration.scala

```diff
@@ -47,10 +50,37 @@ case class ExportsDeclaration(
 
 object ExportsDeclaration {
 
-  object REST {
+  def apply(id: String, eori: Eori, declarationRequest: ExportsDeclarationRequest): ExportsDeclaration =
+    ExportsDeclaration(
+      id = id,
+      eori = eori.value,
+      status = declarationRequest.consignmentReferences.map(_ => DeclarationStatus.DRAFT).getOrElse(DeclarationStatus.INITIAL),
+      createdDateTime = declarationRequest.createdDateTime,
+      updatedDateTime = declarationRequest.updatedDateTime,
+      sourceId = declarationRequest.sourceId,
+      `type` = declarationRequest.`type`,
+      dispatchLocation = Some(DispatchLocation(codeForDispatchLocation(declarationRequest))),
+      additionalDeclarationType = declarationRequest.additionalDeclarationType,
+      consignmentReferences = declarationRequest.consignmentReferences,
+      transport = declarationRequest.transport,
+      parties = declarationRequest.parties,
+      locations = declarationRequest.locations,
+      items = declarationRequest.items,
+      totalNumberOfItems = declarationRequest.totalNumberOfItems,
+      previousDocuments = declarationRequest.previousDocuments,
+      natureOfTransaction = declarationRequest.natureOfTransaction
+    )
+
+  private val countriesFor_CO_declType = List("Jersey", "Guernsey")
 
-    import play.api.libs.json._
+  def codeForDispatchLocation(declRequest: ExportsDeclarationRequest): String =
+    (for {
+      consignee <- declRequest.parties.consigneeDetails
+      address <- consignee.details.address
+      maybeCO <- if (countriesFor_CO_declType.contains(address.country)) Some(SpecialFiscalTerritory) else None
+    } yield maybeCO).getOrElse(OutsideEU)
```

Comment:
I must say I was surprised by this solution and that this logic happens actually in controller.
I'd argue if this is a good place to put this logic, because if you imagine having more conditionality for declaration, this file would hold too much responsibilities.
For the time being, it seems also a bit inefficient, because you need to perform this logic on every update.

I'd suggest to postpone this logic to the model conversion process which happens only once in the Declaration's lifetime. You can find the responsible class [here](https://github.com/hmrc/customs-declare-exports/blob/d4227ed01c405c01d95614496b52299f46e6012d/app/uk/gov/hmrc/exports/services/mapping/declaration/TypeCodeBuilder.scala#L26).
The good thing is that you already have the whole `ExportsDeclaration` provided as parameter, so you'd just need to move this logic there.